### DISCUSSION TITLE
Creating .pdf form file based on already prepared document template in .doc/.docx/.pdf filled with JSON data incoming from UI (frontend)

### DISCUSSION THREAD

**Discussion by arESrv** [phax/ph-pdf-layout#40] [General]
Hi everyone.

It's not strictly an issue, but rather a question.

I have this scenario: there's a form on a frontend I need to fill, validate and send the data to backend as a **JSON**, so backend will be able to use the data to prepare **.pdf** form.

The forms templates were initially printed on paper and filled by hand, but now the organization wants to digitalize it and partially automate filling the template form so result will be a completed **.pdf** file.

I have two forms templates as a MS Word type files (**.docx**) - but since Word can convert **.docx** to **.pdf** so the templates can be also **.pdf** files - which I need to populate with this data I'm getting from UI.

Both **.docx** templates are already formatted, with all needed fields, images, colors .etc placed in particular order - only thing I need to do is just automate copying the template file, read it to get the template's structure, fill the copied template with data (fill the fields inside/outside tables, check checkboxes, circle or underscore "Yes/No" answers in fields outside/inside tables, fill text boxes with id numbers .etc) and convert it to **.pdf**.

Does the **ph-pdf-layout** library can load the already prepared form template of **.doc**/**.docx**/**.pdf** type, as a base for form creation, recognize all the structure of fields available/needed to be filled, fill them with data and prepare the **.pdf** form file?

If it's not that easy, what additional steps or tools alongside using **ph-pdf-layout** I'd need to utilize?
Or maybe even use completely different library?

Thanks!

**Comment by phax** [comment]
Hi @arESrv,
I took the liberty and converted that to an issue.
To answer briefly: no, this is currently not possible - simply because it was never really in scope of this project.
The main activity to do here, is to create a template language, that mimics the functionality of the existing Java classes - that does not really sound to complicated, but it needs to be done :)

Importing data from a Word document is next level - because Word allows far more formatting options than this library supports - so it would only work, if you "rebuild" the template in the to-be-created template language.

I think something like Jasper Reports is more likely to fulfil your needs - but I don't know if they can "load" Word templates....
hth
